### PR TITLE
Added an example on how to edit an embed

### PR DESCRIPTION
**Pull request by Henonicks**
Inspired by a question in our Discord, I thought it's a good idea to explain how to change embeds. Don't know what you think, this is my opinion. ¯\\_(ツ)_/¯

I haven't tested this code at all and that's planned for tomorrow (16.04). But if it works, I just need to make a screenshot and add some text.

## Documentation change checklist

- [x] My documentation changes follow the [docs style guide](https://dpp.dev/docs-standards.html) and any code examples follow the [coding style guide](https://dpp.dev/coding-standards.html).
- [x] I tested that my change works before raising the PR (via running `doxygen`, and testing examples).
- [X] I have not moved any existing pages or changed any existing URLs without strong justification as to why.
- [X] I have not generated content using AI or a desktop utility such as grammarly.

### CODE REVIEW

**Comment by Henonicks** [comment]
Sorry for being inactive, busy with school. I'm pretty sure I can continue on Friday.

File: docpages/example_code/editing_messages.cpp [hunk 1]
```diff
@@ -46,9 +44,38 @@ int main() {
...
+			/* here string will automatically be converted to snowflake */
+
+			bot.message_get(msg_id, event.command.channel_id, [&bot, description, event](const dpp::confirmation_callback_t& callback) {
+				if (callback.is_error()) {
+					event.reply("error");
+					return;
+				}
+				auto message = callback.get<dpp::message>();
+				auto& embeds = message.embeds;
+				/* change the embed description and edit the message itself */
```
Comment (Jaskowicz1) [inline docpages/example_code/editing_messages.cpp:71]: Put this comment directly above the line, like so:

```c++
/* change the embed description and edit the message itself */
embeds[0].set_description(description);
```

File: docpages/example_code/editing_messages.cpp [hunk 2]
```diff
@@ -46,9 +44,38 @@ int main() {
...
+				.set_url("https://dpp.dev/")
+				.set_author("Some author", "https://dpp.dev/", "https://dpp.dev/DPP-Logo.png")
+				.set_description("Creator is <creator name>");
+			event.reply(embed);
+		} else if (event.command.get_command_name() == "embed-edit") {
+			const auto description = std::get<std::string>(event.get_parameter("desc"));
+
+			/* get message to edit its embed after */
+			const dpp::snowflake msg_id = std::get<std::string>(event.get_parameter("msg-id"));
+			/* here string will automatically be converted to snowflake */
```
Comment (Jaskowicz1) [inline docpages/example_code/editing_messages.cpp:62]: If you're going to switch between putting the comments after the line (which btw nobody does, please stop doing this) and putting it before the line, stick to one.

You put it after here, then on line 71 you put the comment before.

**Review by Jaskowicz1 — CHANGES_REQUESTED** [review]

File: docpages/example_code/editing_messages.cpp (hunk 2 above)
Comment (Henonicks) [reply docpages/example_code/editing_messages.cpp:62]: In that case I'll also have to edit the line 18 which, by the way, was approved in the last PR

Comment (Jaskowicz1) [reply docpages/example_code/editing_messages.cpp:62]: > In that case I'll also have to edit the line 18 which, by the way, was approved in the last PR

shouldn't have been, it makes no sense.

File: docpages/example_programs/the_basics/editing-channels-and-messages.md [hunk 3]
```diff
@@ -1,15 +1,23 @@
...
-
-\image html stuff_edit1.png
-
-After editing:
-
-\image html stuff_edit2.png
\ No newline at end of file
+\page editing-channels-and-messages Editing Channels and Messages
+
+Sometimes we need to update an object, such as a message (whether it's plain text or an embed) or a channel. At first, it might seem confusing, but it's actually really simple! You need an object with all of the properties being identical to the existing one. Say you're editing a message. You need to have an object identical to the one in Discord. Then you replace what you need such as its content. NOTE: your bot can't edit messages sent by others.
```
Comment (Jaskowicz1) [inline docpages/example_programs/the_basics/editing-channels-and-messages.md:3]: Stick the `NOTE:` in an actual `\note`.

`You need to have an object identical to the one in Discord` I feel like this is misleading, the only thing that needs to match is the ID, every other field can be changed (which is what you do to edit it).

**Review by Jaskowicz1 — CHANGES_REQUESTED** [review]

File: docpages/example_programs/the_basics/editing-channels-and-messages.md (hunk 3 above)
Comment (Jaskowicz1) [reply docpages/example_programs/the_basics/editing-channels-and-messages.md:3]: Btw, you don't have to correct the `You need to have an object identical to the one in Discord` part here right now, would like to see what others think.

File: docpages/example_code/editing_messages.cpp [hunk 4]
```diff
@@ -46,9 +43,37 @@ int main() {
...
+
+			bot.message_get(msg_id, event.command.channel_id, [&bot, description, event](const dpp::confirmation_callback_t& callback) {
+				if (callback.is_error()) {
+					event.reply("error");
+					return;
+				}
+				auto message = callback.get<dpp::message>();
+				auto& embeds = message.embeds;
+
+				/* change the embed description and edit the message itself.
```
Comment (Jaskowicz1) [inline docpages/example_code/editing_messages.cpp:70]: This doesn't follow our Code Conventions.

Multi-line comments should be made in the following format:
```cpp
/* thing
 * thing two
 * thing three
 */
```

**Review by Jaskowicz1 — CHANGES_REQUESTED** [review]

File: docpages/example_programs/the_basics/editing-channels-and-messages.md (hunk 3 above)
Comment (Henonicks) [reply docpages/example_programs/the_basics/editing-channels-and-messages.md:3]: > Stick the `NOTE:` in an actual `\note`.

Should I move the references to other pages into another line? How do I do so while keeping them in the same note block?

Comment (Jaskowicz1) [reply docpages/example_programs/the_basics/editing-channels-and-messages.md:3]: just do two separate note blocks

Comment (Henonicks) [reply docpages/example_programs/the_basics/editing-channels-and-messages.md:3]: ok

File: docpages/example_code/editing_messages.cpp
Comment (Jaskowicz1) [inline docpages/example_code/editing_messages.cpp:1]: I'm just thinking, looking at the page, this is a LOT of content to just throw all at once and I can 100% guarantee that people will skim over this. You should split this into chunks, talk about editing messages first and show code relevant to that, then do embeds and show code relevant to that.

File: docpages/example_programs/the_basics/editing-channels-and-messages.md [hunk 5]
```diff
@@ -1,15 +1,24 @@
...
+
+Before editing:
+
+\image html stuff_edit1.png
+
+After editing:
+
+\image html stuff_edit2.png
+
+Now let's send the embed:
```
Comment (Jaskowicz1) [inline docpages/example_programs/the_basics/editing-channels-and-messages.md:18]: Yeah, this is where splitting would be better. You're randomly just going "now let's send the embed". Remove the embed code stuff from the messaging section and do a new code block here and talk about editing embeds here.

**Review by Jaskowicz1 — CHANGES_REQUESTED** [review]

File: docpages/example_code/editing_messages.cpp
Comment (Henonicks) [reply docpages/example_code/editing_messages.cpp:1]: I'm gonna be on it in a couple of minutes possibly

File: docpages/example_programs/the_basics/editing-channels-and-messages.md (hunk 5 above)
Comment (Henonicks) [reply docpages/example_programs/the_basics/editing-channels-and-messages.md:18]: Fair

**Comment by Jaskowicz1** [comment]
Any updates?

**Comment by Henonicks** [comment]
> Any updates?

Started working on it not too long ago but didn't have a lotta time. Expect a push pretty soon.

**Comment by Jaskowicz1** [comment]
> > Any updates?
> 
> Started working on it not too long ago but didn't have a lotta time. Expect a push pretty soon.

Sounds good!

**Comment by Henonicks** [comment]
Just ran doxygen

File: docpages/example_programs/the_basics/editing-channels-and-messages.md [hunk 6]
```diff
@@ -1,15 +1,42 @@
...
-\image html stuff_edit2.png
\ No newline at end of file
+\page editing-channels-and-messages Editing Channels and Messages
+
+Sometimes we need to update an object, such as a message (whether it's plain text or an embed) or a channel. At first, it might seem confusing, but it's actually really simple! You need an object with all the properties being identical to the existing one. Say you're editing a message. You need to have an object with its ID the same as the one in Discord. Then you replace what you need, such as its content.
+
+\note Your bot can't edit messages sent by others!
+\note This example uses callback functions and embeds. To see more information about them, visit \ref callback-functions and \ref embed-message.
+
+Here we send a message and edit it after. To do so, we first reply to the command `msg-send` with some text, "This is a message" in our case. As described above, on the next step the message object is taken and the text is replaced with whatever the user desires.
```
Comment (Jaskowicz1) [inline docpages/example_programs/the_basics/editing-channels-and-messages.md:8]: I think you should separate each section here with `## Editing Messages`, `## Editing channels`, etc. Just so, when you're quickly scrolling through looking for something (or when we tell someone to look at this page for a specific reason), they  can quickly find it.

**Review by Jaskowicz1 — CHANGES_REQUESTED** [review]

File: docpages/example_programs/the_basics/editing-channels-and-messages.md [hunk 7]
```diff
@@ -1,15 +1,42 @@
...
+
+Before editing the channel:
+
+\image html stuff_edit3.png
+
+After editing the channel:
+
+\image html stuff_edit4.png
+
+Now let's send the embed and edit it. It's close but a bit more different than before. Say if a message has one `content` field, it can have a few `embed` fields, up to 10 to be precise. So we first get the embed we want and edit and change its description.
```
[the remark this answers is not in the document]
Comment (Henonicks) [reply docpages/example_programs/the_basics/editing-channels-and-messages.md:32]: > You should reword this to talk about how you're now going to cover editing embeds.

Sounds fine to me, just the "the" should be replaced with "an".
About the `## Editing <stuff>`, done

[the remark this answers is not in the document]
Comment (Jaskowicz1) [reply docpages/example_programs/the_basics/editing-channels-and-messages.md]: > Sounds fine to me, just the "the" should be replaced with "an".

yeah that's kinda implied when I said "reword" lol

File: docpages/example_programs/the_basics/editing-channels-and-messages.md [hunk 8]
```diff
@@ -1,15 +1,45 @@
...
+Before editing the channel:
+
+\image html stuff_edit3.png
+
+After editing the channel:
+
+\image html stuff_edit4.png
+
+## Editing embeds
+Now let's send an embed and edit it. It's close but a bit more different than before. Say if a message has one `content` field, it can have a few `embed` fields, up to 10 to be precise. So we first get the embed we want and edit and change its description.
```
Comment (Jaskowicz1) [inline docpages/example_programs/the_basics/editing-channels-and-messages.md:35]: Remove `It's close but a bit more different than before.`

Change `Say if a message...` to `If a message...`

**Review by Jaskowicz1 — CHANGES_REQUESTED** [review]
Nearly there, much better now though!

File: docpages/example_programs/the_basics/editing-channels-and-messages.md [hunk 9]
```diff
@@ -1,15 +1,45 @@
...
+
+Sometimes we need to update an object, such as a message (whether it's plain text or an embed) or a channel. At first, it might seem confusing, but it's actually really simple! You need an object with all the properties being identical to the existing one. Say you're editing a message. You need to have an object with its ID the same as the one in Discord. Then you replace what you need, such as its content.
+
+## Editing messages
+Here we send a message and edit it after. To do so, we first reply to the command `msg-send` with some text, "This is a message" in our case. As described above, on the next step the message object is taken and the text is replaced with whatever the user desires.
+
+\include{cpp} editing_messages1.cpp
+
+\note Your bot can't edit messages sent by others!
+\note This example uses callback functions and embeds. To see more information about them, visit \ref callback-functions and \ref embed-message.
```
Comment (Jaskowicz1) [inline docpages/example_programs/the_basics/editing-channels-and-messages.md:11]: Why did you move the callback one?

Move the callbacks note outside this (like line 4)

**Review by Jaskowicz1 — CHANGES_REQUESTED** [review]

File: docpages/example_programs/the_basics/editing-channels-and-messages.md (hunk 9 above)
Comment (Henonicks) [reply docpages/example_programs/the_basics/editing-channels-and-messages.md]: When you don't read what you wrote this happens

**Review by Jaskowicz1 — APPROVED** [review]
